tika-dl/src/main/java/org/apache/tika/dl/imagerec/DL4JVGG16Net.java

```diff
@@ -0,0 +1,161 @@
+				locationToSave= new File(MODEL_DIR+File.separator+"tikaPreprocessed"+File.separator+"vgg16.zip");
+                    helper = new TrainedModelHelper(TrainedModels.VGG16NOTOP);
+                    break;*/
+                case "VGG16":
+                    helper = new TrainedModelHelper(TrainedModels.VGG16);
+                    modelFile = new File(MODEL_DIR_PREPROCESSED + File.separator + "vgg16.zip");
+                    locationToSave = new File(MODEL_DIR + File.separator + "tikaPreprocessed" + File.separator + "vgg16.zip");
+                    break;
+                default:
+                    throw new TikaConfigException("Unknown or unsupported model");
+            }
+            if (serialize.trim().toLowerCase(Locale.ROOT).equals("yes")) {
+                if (!modelFile.exists()) {
+                    LOG.warn("Preprocessed Model doesn't exist at {}", modelFile);
+                    modelFile.getParentFile().mkdirs();
+                    model = helper.loadModel();
+                    LOG.info("Saving the Loaded model for future use. Saved models are more optimised to consume less resources.");
+                    ModelSerializer.writeModel(model, locationToSave, true);
+                    available = true;
+                } else {
+                    model = ModelSerializer.restoreComputationGraph(locationToSave);
+                    LOG.info("Preprocessed Model Loaded from {}", locationToSave);
+                    available = true;
+                }
+
+            } else if (serialize.trim().toLowerCase(Locale.ROOT).equals("no")) {
+                LOG.info("Weight graph model loaded via dl4j Helper functions");
+                model = helper.loadModel();
+                available = true;
+            } else {
+                throw new TikaConfigException("Configuration Error. serialization can be either yes or no.");
+            }
+
+            if (!available) {
+                return;
+            }
+            HashMap<Pattern, String> patterns = new HashMap<>();
+            patterns.put(Pattern.compile(outPattern), null);
+            setMetadataExtractionPatterns(patterns);
+            setIgnoredLineConsumer(IGNORED_LINE_LOGGER);
+        } catch (Exception e) {
+            LOG.warn("exception occured");
+            throw new TikaConfigException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public List<RecognisedObject> recognise(InputStream stream, ContentHandler handler,
+                                            Metadata metadata, ParseContext context)
+            throws IOException, SAXException, TikaException {
+        NativeImageLoader loader = new NativeImageLoader(224, 224, 3);
+        INDArray image = loader.asMatrix(stream);
+        DataNormalization scaler = new VGG16ImagePreProcessor();
+        scaler.transform(image);
+        INDArray[] output = model.output(false, image);
+        String modelOutput = TrainedModels.VGG16.decodePredictions(output[0]);
```

Comment:
We need use the code in `TrainedModels.VGG16.decodePredictions(...)`  to get `topN` objects